src/llvm-gcroot.cpp

```diff
@@ -171,10 +177,189 @@ class JuliaGCAllocator {
         std::map<BasicBlock*, std::map<frame_register, liveness::id> > &bb_uses,
         std::map<BasicBlock*, SmallBitVector> &regs_used);
     void rearrangeRoots();
+    void lowerHandlers();
 public:
     void allocate_frame();
 };
 
+struct HandlerData {
+    // Pairs of <jl_pop_handle call inst>, number of pops left after popping
+    // this frame.
+    std::vector<std::pair<CallInst*,uint64_t>> leaves;
+    // enters that are directly nested in this frame
+    std::set<CallInst*> nested;
+    std::unique_ptr<std::vector<CallInst*>> parent_vec;
+    CallInst *parent{nullptr};
+    bool processed{false};
+};
+
+void JuliaGCAllocator::lowerHandlers()
+{
+    if (!except_enter_func)
+        return;
+    auto jlenter_func = M.getNamedValue("jl_enter_handler");
+    auto setjmp_func = M.getNamedValue(jl_setjmp_name);
+    // Collect all exception enters
+    std::map<CallInst*,HandlerData> handlers;
+    for (auto &BB: F) {
+        for (auto &I: BB) {
+            auto call = dyn_cast<CallInst>(&I);
+            if (call && call->getCalledValue() == except_enter_func) {
+                handlers[call] = HandlerData();
+            }
+        }
+    }
+
+    if (handlers.empty())
+        return;
+
+    typedef std::map<CallInst*,HandlerData>::iterator hdlr_iter_t;
+    // For each exception enter, compute the life time of the enter, find
+    // the corresponding leaves and collect a list of nested exception frames.
+    // This assumes the exception frames has simple structure. E.g.
+    // there's no recursion and different frames do no share the same leave.
+    std::function<void(hdlr_iter_t)> process_handler = [&] (hdlr_iter_t hdlr) {
```

Comment:
msvc says `c:/projects/julia/src/llvm-gcroot.cpp(221) : error C2039: 'function' : is not a member of 'std'`